ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -636,24 +636,15 @@ private boolean shouldUpload(MessageInfo messageInfo) {
 
   @Override
   public void delete(List<MessageInfo> infos) throws StoreException {
-    // TODO: Remove the duplicate code by calling deleteAsync() method.
-    checkStarted();
-    checkDuplicates(infos);
-
     try {
-      for (MessageInfo msgInfo : infos) {
-        BlobId blobId = (BlobId) msgInfo.getStoreKey();
-        // If the cache has been updated by another thread, retry may be avoided
-        requestAgent.doWithRetries(() -> deleteIfNeeded(blobId, msgInfo.getOperationTimeMs(), msgInfo.getLifeVersion()),
-            "Delete", partitionId.toPathString());
+      for (MessageInfo msg : infos) {
+        cloudDestination.deleteBlob((BlobId) msg.getStoreKey(), msg.getOperationTimeMs(), msg.getLifeVersion(), null);
```

Comment:
are we not doing any retries anymore?

Reply:
the retries are handled by the azure client sdk, so i removed any retries from ambry a long time ago. Now i am just removing the code itself because its just unreadable.